docs/tutorials/intro/inventory.rst

```diff
@@ -1,2 +1,131 @@
+
+Finally, the host behaves like a python dictionary so you can iterate over the data as such::
+
+	>>> h.keys()
+	dict_keys(['name', 'group', 'asn', 'vlans', 'site', 'role', 'brigade_nos', 'type'])
+	>>> h.values()
+	dict_values(['host1.cmh', 'cmh', 65000, {100: 'frontend', 200: 'backend'}, 'cmh', 'host', 'linux', 'host'])
+	>>> h.items()
+	dict_items([('name', 'host1.cmh'), ('group', 'cmh'), ('asn', 65000), ('vlans', {100: 'frontend', 200: 'backend'}), ('site', 'cmh'), ('role', 'host'), ('brigade_nos', 'linux'), ('type', 'host')])
+	>>> for k, v in h.items():
+	...     print(k, v)
+	...
+	name host1.cmh
+	group cmh
+	asn 65000
+	vlans {100: 'frontend', 200: 'backend'}
+	site cmh
+	role host
+	brigade_nos linux
+	type host
+	>>>
+
+.. note:: You can head to :obj:`brigade.core.inventory.Host` and :obj:`brigade.core.inventory.Group` for details on all the available attributes and functions for each ``host`` and ``group``.
+
+Filtering the inventory
+-----------------------
+
+You won't always want to operate over all hosts, sometimes you will want to operate over some of them based on some attributes. In order to do so the inventory can help you filtering based on it's attributes. For instance::
+
+	>>> inventory.hosts.keys()
+	dict_keys(['host1.cmh', 'host2.cmh', 'spine00.cmh', 'spine01.cmh', 'leaf00.cmh', 'leaf01.cmh', 'host1.bma', 'host2.bma', 'spine00.bma', 'spine01.bma', 'leaf00.bma', 'leaf01.bma'])
+	>>> inventory.filter(site="bma").hosts.keys()
+	dict_keys(['host1.bma', 'host2.bma', 'spine00.bma', 'spine01.bma', 'leaf00.bma', 'leaf01.bma'])
+	>>> inventory.filter(site="bma", role="spine").hosts.keys()
+	dict_keys(['spine00.bma', 'spine01.bma'])
+	>>> inventory.filter(site="bma").filter(role="spine").hosts.keys()
+	dict_keys(['spine00.bma', 'spine01.bma'])
+
+Note in the last line that the filter is cumulative so you can do things like this:
+
+	>>> cmh = inventory.filter(site="cmh")
+	>>> cmh.hosts.keys()
+	dict_keys(['host1.cmh', 'host2.cmh', 'spine00.cmh', 'spine01.cmh', 'leaf00.cmh', 'leaf01.cmh'])
+	>>> cmh_eos = cmh.filter(brigade_nos="eos")
+	>>> cmh_eos.hosts.keys()
+	dict_keys(['spine00.cmh', 'leaf00.cmh'])
+	>>> cmh_eos.filter(role="spine").hosts.keys()
+	dict_keys(['spine00.cmh'])
+
+This should give you enough room to build groups in any way you want.
+
+Advanced filtering
+__________________
+
+You can also do more complex filtering by using functions or lambdas::
+
+	>>> def has_long_name(host):
+	...     return len(host.name) == 11
+	...
+	>>> inventory.filter(filter_func=has_long_name).hosts.keys()
+	dict_keys(['spine00.cmh', 'spine01.cmh', 'spine00.bma', 'spine01.bma'])
+	>>> inventory.filter(filter_func=lambda h: len(h.name) == 9).hosts.keys()
+	dict_keys(['host1.cmh', 'host2.cmh', 'host1.bma', 'host2.bma'])
+
+Not the most useful example but it should be enough to illustrate how it works.
+
+
+Hosts/Groups contents
```

Comment:
I think it would be helpful if this was displayed before the inventory was loaded. I think it makes it easier to understand what data we are looking at.